src/sprite/sprite.cpp

```diff
@@ -124,7 +124,8 @@ Sprite::update()
   }
 
   while (m_frameidx >= get_frames() && !animation_done()) {
-    m_frameidx -= get_frames();
+    // Loop animation.
+    m_frameidx -= get_frames() - (m_action->loop_frame > 1 ? m_action->loop_frame - 1 : 0);
```

Comment:
Since `loop_frame` is set to 1 if it isn't set in the sprite file, couldn't we all get rid of the whole if statement here?

m_action->loop_frame defaults to 1 which means that both cases of the if statement would result to 0.